### ISSUE TITLE
[BUG-Mac] Crash when exiting/quit. Cannot interact with menus after

### ISSUE THREAD

**Issue by reaperblog** [obsproject/obs-studio#9929]
### Operating System Info

macOS 14

### OBS Studio Version

30.0.0

### OBS Studio Log URL

https://obsproject.com/logs/UW3lKgdOSiSKgENH

### OBS Studio Crash Log URL

https://www.dropbox.com/scl/fi/mvt89rv9clora4cmm8d1y/OBS-2023-11-27-160154.ips?rlkey=kwwgouk6chrbsfyrszbv1l74o&dl=0

### Expected Behavior

The program should exit successfully.

### Current Behavior

If I use CMD-Q, File - Quit, control section - Exit, or Help - restart in safe/normal mode. The app exits along with a Mac crash log.
Additionally, and I don't know if this is the same bug but it occurs after the crash…
When reopening OBS I can't interact with the + buttons to add scenes or sources. I can't right click on anything. Properties or filter windows will not open.

### Steps to Reproduce

1. Open OBS
2. CMD-Q, File - Quit, control section - Exit, or Help - restart in safe/normal mode.
3.
...

**Comment by jcm93** [comment]
It might help to also upload a crash log from exiting OBS while it's in safe mode; this would rule out any side effects from plugins.

**Comment by reaperblog** [comment]
I should say the crash isn't every time, 3 out of 4 times. 

ok this report is from safe mode
https://www.dropbox.com/scl/fi/xq344dox14olwjcywr4vx/OBS-2023-11-27-183416.ips?rlkey=ia2wym11sq7xg1h1cltqq6irk&dl=0

I also just noticed that every time OBS crashes at exit, iOSScreenCaptureAssistant simultaneously crashes.
https://www.dropbox.com/scl/fi/nkhqu4f9rmcu57cmjmxia/iOSScreenCaptureAssistant-2023-11-27-183415.ips?rlkey=f8mlj0y6kl1e6yqgaio6d74sx&dl=0

**Comment by RytoEX** [comment]
The crash appears to be due to an issue with the YouTube stats dock.

The UI being unresponsive sounds more like #9970, where the UI itself is responsive, but child windows (and maybe menus/submenus?) do not spawn in a visible location. A video of this behavior might help.

**Comment by reaperblog** [comment]
> The crash appears to be due to an issue with the YouTube stats dock.
> 
> The UI being unresponsive sounds more like #9970, where the UI itself is responsive, but child windows (and maybe menus/submenus?) do not spawn in a visible location. A video of this behavior might help.

yes, just like the video posted there. for right-click menu I do believe it is there but invisible because a left click where the menu should be will hit a menu item.
When I stream tomorrow I will try with the stats dock removed.

**Comment by peancored** [comment]
I can confirm that it does hit the menu item for me upon clicking an invisible menu item, too.

**Comment by reaperblog** [comment]
> The crash appears to be due to an issue with the YouTube stats dock.
> 
This is it. "Youtube Live Control Panel" Dock.
I hid that dock and haven't had the crash or the menu problem. No issue with my stream last week, and just now I opened and closed OBS several times as a test.

**Comment by snicolai** [comment]
I had a crash when quitting OBS 30.0.2 with both the Youtube Live Control Panel dock open and the Youtube chat dock open.  MacOS 14.2.1 (23C71).  I had streamed for about 20 minutes, but stopped the stream before quitting.

[OBS Crash on quit.txt](https://github.com/obsproject/obs-studio/files/13747468/OBS.Crash.on.quit.txt)

**Comment by huegrowth** [comment]
I do not have any youtube panel or docks but still encounter the same pbm.

**Comment by huegrowth** [comment]
Ok found the problem came from AITUM Vertical plugin. Downloaded the latest version. And it's good

**Comment by jcm93** [comment]
Users have also reported that OBS crashes upon being minimized, so the issue seems possibly related to these docks no longer being visible, rather than being completely terminated. Seems to occur with the YouTube Live Control Panel dock, as well as some docks from at least one other service (Lovense), per user reports.

Same crash on Linux (presumably): https://github.com/obsproject/obs-studio/issues/10011
